src/EA.Weee.RequestHandlers.Tests.Unit/AatfReturn/ReturnSchemeDataAccessTests.cs

```diff
@@ -80,5 +80,48 @@ public async Task GetOperatorByReturnId_GivenReturnIdDoesNotExist_NullShouldBeRe
 
             result.Should().BeNull();
         }
+
+        [Fact]
+        public async Task RemoveReturnScheme_GivenSchemeIdReturnId_PCSforReturnShouldBeReturned()
+        {
+            var returnId = Guid.NewGuid();
+            var schemeId1 = Guid.NewGuid();
+            var schemeIds = new List<Guid>();
+            schemeIds.Add(schemeId1);
+
+            var returnSchemeMatch = A.Fake<ReturnScheme>();
+            var returnSchemeNoMatch = A.Fake<ReturnScheme>();
+
+            A.CallTo(() => returnSchemeMatch.Scheme).Returns(A.Fake<Scheme>());
+            A.CallTo(() => returnSchemeMatch.SchemeId).Returns(schemeId1);
+            A.CallTo(() => returnSchemeMatch.ReturnId).Returns(returnId);
+
+            A.CallTo(() => returnSchemeNoMatch.Scheme).Returns(A.Fake<Scheme>());
+            A.CallTo(() => returnSchemeNoMatch.SchemeId).Returns(schemeId1);
+            A.CallTo(() => returnSchemeNoMatch.ReturnId).Returns(A.Dummy<Guid>());
+
+            List<WeeeReceived> weeeReceived = new List<WeeeReceived>();
+
+            WeeeReceived weee = new WeeeReceived(schemeId1, A.Dummy<Guid>(), returnId);           
+            WeeeReceived weee1 = new WeeeReceived(schemeId1, A.Dummy<Guid>(), A.Dummy<Guid>());
+            weeeReceived.Add(weee);
+            weeeReceived.Add(weee1);
+
+            var list = new List<WeeeReceivedAmount>();
+            var weeeReceivedAmount = A.Fake<WeeeReceivedAmount>();
+
+            A.CallTo(() => weeeReceivedAmount.Id).Returns(A.Dummy<Guid>());
+            list.Add(weeeReceivedAmount);
+            A.CallTo(() => context.WeeeReceivedAmount).Returns(dbHelper.GetAsyncEnabledDbSet(list));
+
+            A.CallTo(() => context.WeeeReceived).Returns(dbHelper.GetAsyncEnabledDbSet(weeeReceived));
+            A.CallTo(() => context.ReturnScheme).Returns(dbHelper.GetAsyncEnabledDbSet(new List<ReturnScheme>() { returnSchemeNoMatch, returnSchemeMatch }));
+
+            await dataAccess.RemoveReturnScheme(schemeIds, returnId);
+
+            A.CallTo(() => context.WeeeReceived.Remove(weee)).MustHaveHappened(Repeated.Exactly.Once);
```

Comment:
Can we also check that the return that does not match is still in the context

Reply:
Added